File: trunk/EngineUAB/Run/Videogame/Data/Scripts/Game/Player/player_animation_attack_state.lua
class 'CPlayerAnimationAttackState' (CState)
	function CPlayerAnimationAttackState:__init(name) 
		CState.__init(self, name)
	end

	function CPlayerAnimationAttackState:OnEnter(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("idle")
			_CCharacter:get_animation_model():blend_cycle( num, 0.1 )
			
			num = _CCharacter:get_animation_id("attack1")
			_CCharacter:get_animation_model():execute_action( num, 0.1 )
		end
	end
	
	function CPlayerAnimationAttackState:Execute(_CCharacter, _elapsed_time)
	end
	
	function CPlayerAnimationAttackState:OnExit(_CCharacter)
		num = _CCharacter:get_animation_id("idle")
		_CCharacter:get_animation_model():clear_cycle( num, 0.3 )
	end
	
	function CPlayerAnimationAttackState:OnMessage(_CCharacter, _Msg)
	
		return false
	end
	
	function CPlayerAnimationAttackState:__Finalize()
	end

File: EngineUAB/Code/Videogame/Characters/Player/AnimationStates/PlayerAnimationAttackState.cpp
#include "PlayerAnimationAttackState.h"
#include "StatesMachine\Telegram.h"
#include "RenderableObjects\AnimatedModel\AnimatedInstanceModel.h"

#if defined _DEBUG
#include "Memory\MemLeaks.h"
#endif

CPlayerAnimationAttackState::CPlayerAnimationAttackState( CCharacter* _pCharacter,const std::string &_Name )
	: CState(_pCharacter, _Name)
{

}

CPlayerAnimationAttackState::~CPlayerAnimationAttackState()
{

}

void CPlayerAnimationAttackState::OnEnter( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("attackfinal1");
	_pCharacter->GetAnimatedModel()->BlendCycle(l_iAnimID, 0.1f);

	l_iAnimID = _pCharacter->GetAnimationID("attack1");
	_pCharacter->GetAnimatedModel()->ExecuteAction(l_iAnimID, 0.1f);
}

void CPlayerAnimationAttackState::Execute( CCharacter* _pCharacter, float _fElapsedTime )
{

}

void CPlayerAnimationAttackState::OnExit( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("attackfinal1");
	_pCharacter->GetAnimatedModel()->ClearCycle(l_iAnimID, 0.1f);
}

bool CPlayerAnimationAttackState::OnMessage( CCharacter* _pCharacter, const STelegram& _Message )
{
	return false;
}